src/app/item-page/edit-item-page/item-status/item-status.component.ts

```diff
@@ -72,12 +91,37 @@ export class ItemStatusComponent implements OnInit {
         lastModified: item.lastModified
       });
       this.statusDataKeys = Object.keys(this.statusData);
+
+      // Observable for item identifiers (retrieved from embedded link)
+      this.identifiers$ = this.identifierDataService.getIdentifierDataFor(item).pipe(
+        map((identifierRD) => {
+          if (identifierRD.statusCode !== 401 && hasValue(identifierRD.payload)) {
+            return identifierRD.payload.identifiers;
+          } else {
+            return null;
+          }
+        }),
+      );
+
+      // Observable for configuration determining whether the Register DOI feature is enabled
+      let registerConfigEnabled$: Observable<boolean> = this.configurationService.findByPropertyName('identifiers.item-status.register-doi').pipe(
```

Comment:
@atarix83 I request the property to determine if the button should be drawn at all, then in the pipe where features are checked, the DOI button is added first (if applicable), so it is also checked as a feature (just like all the other operations in the array) to set the enabled/disabled state.

I wanted these to both apply, so that people who chose not to use this new functionality at all weren't bothered by an irrelevant button. If the config is enabled but the user lacks authorization for some reason or other, then it will appear but be disabled.

Does that make sense?

Reply:
it makes sense, thanks